ios/RnCryptoTests/AesCipherTest.swift

```diff
@@ -6,77 +6,286 @@
+    let destination = NSTemporaryDirectory() + "/encrypt"
+    let inputStream = InputStream.init(data: "teststring".data(using: .utf8)!)
+    let outputStream = OutputStream(url: URL(fileURLWithPath: destination), append: true)
+
+    let hexKey = "4ba9058b2efc8c7c9c869b6573b725aa8bf67aecb26d3ebd678e624565570e9c"
+
+    self.sut.encrypt(
+      input: inputStream,
+      output: outputStream!,
+      key: self.utils.hexStringToBytes(hexKey),
+      iv: [UInt8](),
+      callback: { (error, status) in
+        XCTAssertEqual(status, nil)
+        XCTAssertEqual(error, RnCryptoError.badIv)
+
+      }
+    )
+  }
+
+  // Test encryption to chunks with unaligned buffer and chunk sizes
+  func testBufferOverflowWithUnalignedChunks() {
+    // Testing with buffer size of 8KB and non-multiple chunk size of 7KB
+    let testData = Data(repeating: 0x41, count: 16 * 1024)
+    let chunkSize = 7 * 1024
+
+    let outputs = [
+      OutputStream(toMemory: ()),
+      OutputStream(toMemory: ()),
+      OutputStream(toMemory: ()),
+    ]
+
+    let expectation = XCTestExpectation(description: "Unaligned chunks test")
+
+    sut.encryptToChunks(
+      input: InputStream(data: testData), outputs: outputs,
+      key: validKey, iv: validIV, chunkSize: chunkSize
+    ) { error, status in
+      XCTAssertNil(error)
+
+      let chunks = outputs.map { $0.property(forKey: .dataWrittenToMemoryStreamKey) as! Data }
+
+      XCTAssertEqual(chunks[0].count, 7168)  // First 7KB chunk
+      XCTAssertEqual(chunks[1].count, 7168)  // Second 7KB chunk
+      XCTAssertEqual(chunks[2].count, 2048)  // Remaining 2KB
+
+      XCTAssertEqual(chunks.reduce(0) { $0 + $1.count }, testData.count)
+      expectation.fulfill()
     }
+
+    wait(for: [expectation], timeout: 5.0)
+  }
+
+  func testEncryptToChunks() {
+      let testData = Array(repeating: UInt8(1), count: 100)
+      let key = Array(repeating: UInt8(0), count: 32)
+      let iv = Array(repeating: UInt8(0), count: 16)
+      let partSize = 30
+
+      let inputStream = InputStream(data: Data(testData))
+      let outputStreams = [
+          OutputStream(toMemory: ()),
+          OutputStream(toMemory: ()),
+          OutputStream(toMemory: ()),
+          OutputStream(toMemory: ())
+      ]
+      
+      inputStream.open()
+      outputStreams.forEach { $0.open() }
+      
+      let expectation = XCTestExpectation(description: "Encryption test")
+      var encryptError: RnCryptoError?
+
+      sut.encryptToChunks(input: inputStream, outputs: outputStreams, key: key, iv: iv, chunkSize: partSize) { error, status in
+          print("Input status:", inputStream.streamStatus.rawValue)
+          print("Output status:", outputStreams.map { $0.streamStatus.rawValue })
+          print("Error:", error ?? "nil")
+          encryptError = error
+          expectation.fulfill()
+      }
+
+      wait(for: [expectation], timeout: 5.0)
+      XCTAssertNil(encryptError)
+  }
+  
+  func testLargeEncryptToChunks() {
```

Comment:
Unless you have some way of testing the RAM consumption over the encrypting and it is the main objective of this test, this is unnecessary. 
It's preferable to have a test for: 
- When the `partSize` is > than the file size (you should throw, this is not expected)
- When the `partSize` is < than the file size 
   - When the `partSize` is < than the file size and is not multiple of the file size
   - When the `partSize` is < than the file size and is multiple of the file size
- When the `partSize` equals the file size

_You must assert in all of them that the sent content is the same as the provided (by joining the parts content and decrypting, for instance)_ 

Apart of that:
- Any other test to ensure the errors throw when you expect them to throw

Reply:
Added, but this case `When the partSize is > than the file size (you should throw, this is not expected)` instead of throwing error let it work. 
I don't see this behaviour wrong, if we want to throw error I change the implementation of the function.

Reply:
You should not use multipart uploads for that case, it will fail. That's why you should control it in advance.

Reply:
To add this validation here we would have to read the file and I don't think that in terms of performance it would be convenient. 
The other alternative I can think of is to pass the file size to the function, but for that we have to trust that the given file size is correct.

Reply:
But the chunkSize has to be < than the file size. I think I don't understand what you are saying